service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,117 @@
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle());
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents_socialWorkOther", listOfOtherDocs);
+        updatedCaseData.put("documents_socialWorkChronology_document", socialWorkChronologyDocument);
+        updatedCaseData.put("documents_socialWorkStatement_document", socialWorkStatementDocument);
+        updatedCaseData.put("documents_socialWorkAssessment_document", socialWorkAssessmentDocument);
+        updatedCaseData.put("documents_socialWorkCarePlan_document", socialWorkCarePlanDocument);
+        updatedCaseData.put("documents_socialWorkEvidenceTemplate_document", socialWorkEvidenceTemplateDocument);
+        updatedCaseData.put("documents_threshold_document", thresholdDocument);
+        updatedCaseData.put("documents_checklist_document", checklistDocument);
+        updatedCaseData.put("courtBundle", courtBundleDocument);
+
+        return updatedCaseData;
+    }
+
+    public <T extends DocumentMetaData> List<Element<T>> setUpdatedByAndDateAndTimeOnDocuments(
+        List<Element<T>> currentDocuments,
+        List<Element<T>> previousDocuments) {
+
+        Predicate<Element<T>> doesNotContainInOldDocs =
+            doc -> previousDocuments != null && !previousDocuments.contains(doc);
+
+        currentDocuments.stream()
+            .filter(doesNotContainInOldDocs)
+            .forEach(doc -> findElement(doc.getId(), previousDocuments)
+                .ifPresent(e -> {
+                    if (!e.getValue().getTypeOfDocument().equals(doc.getValue().getTypeOfDocument())) {
+                        setUpdatedByAndDateTime(doc);
+                    }
+                })
+            );
+
+        currentDocuments.stream()
+            .filter(doc -> doc.getValue().getDateTimeUploaded() == null)
+            .forEach(this::setUpdatedByAndDateTime);
+
+        return currentDocuments;
+    }
+
+    public <T extends DocumentMetaData> T setUpdatedByAndDateAndTimeOnDocuments(T currentDocuments,
+                                                                                T previousDocuments) {
+        if (currentDocuments == null || currentDocuments.getTypeOfDocument() == null) {
+            return null;
+        }
+
+        if ((previousDocuments != null
+            && !currentDocuments.getTypeOfDocument().equals(previousDocuments.getTypeOfDocument()))
+            || currentDocuments.getDateTimeUploaded() == null) {
+            return setUpdatedByAndDateTime(currentDocuments);
+        }
+        return null;
+    }
+
+    private <T extends DocumentMetaData> T setUpdatedByAndDateTime(T document) {
+        String uploadedBy = documentUploadHelper.getUploadedDocumentUserDetails();
+
+        document.setDateTimeUploaded(time.now());
+        document.setUploadedBy(uploadedBy);
+        return document;
+    }
+
+    private <T extends DocumentMetaData> void setUpdatedByAndDateTime(Element<T> doc) {
```

Comment:
param should really be `documentElement` but consider if we can remove this overload.

Reply:
I think doc is fine however it should just call the above method and ignore the return value.

Reply:
or yeah it could be removed all together

Reply:
renamed

Reply:
removed method.